openedx/core/lib/request_utils.py

```diff
@@ -103,162 +90,102 @@ def course_id_from_url(url):
         return None
 
 
-class CookieMonitoringMiddleware(MiddlewareMixin):
+class CookieMonitoringMiddleware:
     """
     Middleware for monitoring the size and growth of all our cookies, to see if
     we're running into browser limits.
     """
-    def process_request(self, request):
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        # monitor after response to ensure logging can include user id where possible.
+        self.log_and_monitor_cookies(request)
```

Comment:
Would it make sense to put this behind a try block that just emits a warning on exception?

Reply:
Yes. Done.